dashboard/app/models/peer_review.rb

```diff
@@ -30,14 +30,53 @@ class PeerReview < ActiveRecord::Base
   belongs_to :level
   belongs_to :level_source
 
+  after_save :mark_user_level
+
   REVIEWS_PER_SUBMISSION = 2
+  REVIEWS_FOR_CONSENSUS = 2
 
   enum status: {
     accepted: 0,
     rejected: 1,
     escalated: 2
   }
 
+  def mark_user_level
+    user_level = UserLevel.find_by!(user: submitter, level: level)
+
+    # Instructor feedback should override all other feedback
+    if from_instructor
+      user_level.update!(best_result: accepted? ? Activity::REVIEW_ACCEPTED_RESULT : Activity::REVIEW_REJECTED_RESULT)
+      return
+    end
+
+    # Ignore negative peer feedback after a submission has already been approved
+    return if user_level.best_result == Activity::REVIEW_ACCEPTED_RESULT
+
+    # Only look at reviews for the most recent submission
+    most_recent = submitter.last_attempt(level).try(:level_source_id)
+    return unless level_source_id == most_recent
+
+    # Find all current PeerReviews for this submission
+    reviews = PeerReview.where(
+      submitter: submitter,
+      script: script,
+      level: level,
+      level_source_id: most_recent
+    ).where.not(status: nil)
```

Comment:
Hmm but you're right that I can re-write the next lines to use the enum scopes instead of `.all?`.